src/controllers/character-image-controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Express, Request, Response } from 'express';
+
+import { CharacterImage } from '../models/character-image';
+import { requestImageThumbnail } from './async-controller';
+import { errorServer, errorInvalidBody, successResponse } from '../utils/responses-helper';
+
+export async function addCharacterImage(req: Request, res: Response): Promise<void> {
+    if (req.file === undefined) {
+        errorInvalidBody(res);
+        return;
+    }
+
+    if (req.body["pc_id"] === undefined) {
+        errorInvalidBody(res);
+        return;
+    }
+
+    // create an image object
+    const file: Express.Multer.File = req.file;
+    try {
+        const image: CharacterImage = new CharacterImage({
+            owner_id: 0,
+            pc_id: req.body["pc_id"],
+            image_name: file.originalname,
+            content_type: file.mimetype,
+            image_data: file.buffer,
+            thumbnail_data: null
+        });
```

Comment:
Is there a possibility that the data will be incorrect at this point? If so, use `isValid()` to verify.

Reply:
No, there isn't a chance the data is incorrect here. `image_name`, `content_type`, `image_data` and `thumbnail_data` are only invalid if they are `undefined`. Multer guarantees the `originalname`, `mimetype`, and `buffer` fields exist though. `pc_id` is verified to exist above but I don't verify it is not negative. I'll add a check for that.